felix/daemon/daemon.go

```diff
@@ -457,11 +474,12 @@ configRetry:
 	var syncer Startable
 	var typhaConnection *syncclient.SyncerClient
 	syncerToValidator := calc.NewSyncerCallbacksDecoupler()
-	if typhaAddr != "" {
+
+	if len(typhaAddresses) > 0 {
```

Comment:
I think so long as we Fatal() up above when typhaEnabled && filteredAddress == 0, this can stay how it is (i.e., reaching this branch means typha is disabled)

Reply:
Added commit here [`5baf4ee` (#5845)](https://github.com/projectcalico/calico/pull/5845/commits/5baf4ee6645049e292e6d5e8de5e16555bfd726c) that I think should address it?

Reply:
I restructured this yesterday a little.  Basically we should never filter out ALL of the addresses anymore.  The bootstrap&filter function will remove wireguard if ALL of the typha endpoints would be filtered out and then it will return all of the typha endpoints.

If there are zero typha endpoints at this point then typha is not configured (since discovery would return a not ready error if typha was enabled but not available).